src/main/java/zaasclient/HttpsClient.java

```diff
@@ -0,0 +1,201 @@
+    private String trustStorePath;
+    private String keyStorePath;
+
+    public HttpsClient(ConfigProperties configProperties) {
+        this.requestConfig = this.buildCustomRequestConfig();
+        this.keyStorePath = configProperties.getKeyStorePath();
+        this.keyStorePassword = configProperties.getKeyStorePassword();
+        this.keyStoreType = configProperties.getKeyStoreType();
+        this.trustStorePath = configProperties.getTrustStorePath();
+        this.trustStorePassword = configProperties.getTrustStorePassword();
+        this.trustStoreType = configProperties.getTrustStoreType();
+    }
+
+
+    public CloseableHttpClient getHttpsClientWithTrustStore()
+        throws NoSuchAlgorithmException, KeyStoreException, IOException, CertificateException, KeyManagementException {
+        SSLContext sslContext;
+        TrustManagerFactory tmf = null;
+
+        if (trustStorePath != null) {
+            tmf = TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
+            KeyStore trustStore = KeyStore.getInstance(trustStoreType);
+            File trustFile = new File(trustStorePath);
+            trustStore.load(new FileInputStream(trustFile), trustStorePassword.toCharArray());
+            tmf.init(trustStore);
+        }
+        sslContext = this.getSSLContext(null, tmf);
+
+        return getCloseableHttpClient(sslContext);
+    }
+
+    /**
+     * an overloaded version of getHttpsClientWithTrustStore for the query api
+     *
+     * @param cookieStore
+     * @return
+     * @throws NoSuchAlgorithmException
+     * @throws KeyStoreException
+     * @throws IOException
+     * @throws CertificateException
+     * @throws KeyManagementException
+     */
+    public CloseableHttpClient getHttpsClientWithTrustStore(BasicCookieStore cookieStore)
+        throws NoSuchAlgorithmException, KeyStoreException, IOException, CertificateException, KeyManagementException {
+        SSLContext sslContext;
+        TrustManagerFactory tmf = null;
+
+        if (trustStorePath != null) {
+            tmf = TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
+            KeyStore trustStore = KeyStore.getInstance(trustStoreType);
+            File trustFile = new File(trustStorePath);
+            trustStore.load(new FileInputStream(trustFile), trustStorePassword.toCharArray());
+            tmf.init(trustStore);
+        }
+        sslContext = this.getSSLContext(null, tmf);
+
+        return getCloseableHttpClient(sslContext, cookieStore);
+    }
+
+    /**
+     * This method is used to return httpClient with TrustStore & KeyStore
+     *
+     * @return
+     * @throws NoSuchAlgorithmException
+     * @throws IOException
+     * @throws KeyStoreException
+     * @throws CertificateException
+     * @throws UnrecoverableKeyException
+     * @throws KeyManagementException
+     */
+    public CloseableHttpClient getHttpsClientWithKeyStoreAndTrustStore()
```

Comment:
Same as above. We can use the extracted method from the common blocks, and similarly we can create a method to create the KeyManagerFactory.